go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -551,6 +579,23 @@ func (dp *diffPartitions) Next() (sql.Partition, error) {
 		}
 
 		if next != nil {
+			// We want to render the dolt_diff table from a top down commit log traversal until we reach a point
+			// where we cannot diff two commits due to a schema difference. This logic is similar to whats in
+			// dp.getRowIter() for two reasons.
+			// 1. The dolt_commit_diff table exclusively uses getRowIter and not the diffPartitions object
+			// 2. The GMS partition exchange does not process partitions in the order they are sent in. This can allow
+			// commits with the computable diffs before the schema change to be rendered, convoluting the UX of the diff
+			// table.
```

Comment:
It's a good idea to document this facet of the diff table, but this comment is too implementation-oriented and won't make sense to the next person reading this code. See comment suggestions.

Reply:
```suggestion
// if we can't diff this commit with its parent, don't traverse any lower
```